src/decaylanguage/dec/dec.py

```diff
@@ -684,41 +705,75 @@ def print_decay_modes(
         print_model: bool, optional, default=True
             Specify whether to print the decay model and model parameters,
             if available.
+        display_photos_keyword: bool, optional, default=True
+            Display the "PHOTOS" keyword in decay models.
         ascending: bool, optional, default=False
             Print the list of decay modes ordered in ascending/descending order
             of branching fraction.
+        normalize: bool, optional, default=True
+            Print the branching fractions normalized to unity
+            (this does not affect the values parsed and actually stored in memory).
         """
         if pdg_name:
             mother = PDG2EvtGenNameMap[mother]
 
         dms = self._find_decay_modes(mother)
 
-        ls = []
+        ls_dict = dict()
         for dm in dms:
-            if print_model:
-                dm_details = self._decay_mode_details(dm)
-                ls.append(
-                    (
-                        dm_details[0],
-                        (
-                            "%-50s %15s %s"
-                            % (
-                                "  ".join(dm_details[1]),
-                                dm_details[2],
-                                dm_details[3],
-                            )
-                        ),
-                    )
-                )
+            bf, fsp_names, model, model_params = self._decay_mode_details(
+                dm, display_photos_keyword
+            )
+            model_params = [str(i) for i in model_params]
+            ls_dict[bf] = (fsp_names, model, model_params)
 
-            else:
-                fsp_names = get_final_state_particle_names(dm)
-                ls.append((get_branching_fraction(dm), "%-50s" % "  ".join(fsp_names)))
+        dec_details = list(ls_dict.values())
+        ls_attrs_aligned = list(
+            zip_longest(
+                *[self._align_items(i) for i in zip(*dec_details)], fillvalue=""
+            )
+        )
 
+        ls = [(bf, ls_attrs_aligned[idx]) for idx, bf in enumerate(ls_dict)]
         ls.sort(key=operator.itemgetter(0), reverse=(not ascending))
+        norm = sum(bf for bf, _ in ls) if normalize else 1
 
         for bf, info in ls:
-            print("{:12g} : {}".format(bf, info))
+            if print_model:
+                line = "  {:.4f}   {}     {}  {}".format(bf / norm, *info)
+            else:
+                line = "  {:.4f}   {}".format(bf / norm, info[0])
+            print(line.rstrip() + ";")
+
+    @staticmethod
+    def _align_items(to_align, align_mode="left", sep=" "):
+        """
+        Left or right align all strings in a list to the same length.
+        By default the string is space-broke into sub-strings and each sub-string aligned individually.
+        """
```

Comment:
Hey, there seems to be a single thing missing, which I hadn't spotted before: it would still be worth having a docstring here and provide the valid values of align_mode, so something like `align_mode: {"left", "right"}, optional, default="left"` together with the rest of the text/explanations. Advance thanks.

Reply:
Done.

Reply:
Thanks.